src/Raven.Server/Documents/AttachmentsStorage.cs

```diff
@@ -487,17 +468,20 @@ public void PutAttachmentStream(DocumentsOperationContext context, Slice key, Sl
             var tree = context.Transaction.InnerTransaction.CreateTree(AttachmentsSlice);
             var existingStream = tree.ReadStream(base64Hash);
             if (existingStream == null)
+            {
                 tree.AddStream(base64Hash, stream, tag: key);
-
+                UpdateBucketStatsOnPutOrDeleteStream(context, key, stream.Length);
+            }
             _documentDatabase.Metrics.Attachments.BytesPutsPerSec.MarkSingleThreaded(stream.Length);
         }
 
-        private void DeleteAttachmentStream(DocumentsOperationContext context, Slice hash, int expectedCount = 1)
+        private void DeleteAttachmentStream(DocumentsOperationContext context, Slice hash, Slice key, int expectedCount = 1)
         {
             if (GetCountOfAttachmentsForHash(context, hash) == expectedCount)
             {
                 var tree = context.Transaction.InnerTransaction.CreateTree(AttachmentsSlice);
-                tree.DeleteStream(hash);
+                var size = tree.DeleteStream(hash).Size;
```

Comment:
For readability let's change it to something like:

```
var deleteResult = tree.DeleteStream(hash);
var size = deleteResult .Size;
```

Reply:
done